tests/stream.rs

```diff
@@ -263,3 +266,90 @@ async fn close_succeeds_if_only_fin_ack_dropped() {
         }
     };
 }
+
+// Test that data is delivered successfully, even if the original SYN-ACK is dropped
+//
+// At the time of writing, a bug in this scenario causes the first bytes (2048 of them) to be
+// silently lost. The recipient thinks it received everything, but is missing the first bytes of
+// the transfer. When the recipient, who started the connection, times out the original SYN, it
+// resends. The sender has already sent some data. When the bug is active, the resent STATE
+// (SYN-ACK) packet uses the sequence number after incrementing from the previously sent state
+// data. This causes the recipient to ignore all data sent previously.
```

Comment:
```suggestion
/// Test that data is delivered successfully, even if the original SYN-ACK is dropped
///
/// At the time of writing, a bug in this scenario causes the first bytes (2048 of them) to be
/// silently lost. The recipient thinks it received everything, but is missing the first bytes of
/// the transfer. When the recipient, who started the connection, times out the original SYN, it
/// resends. The sender has already sent some data. When the bug is active, the resent STATE
/// (SYN-ACK) packet uses the sequence number after incrementing from the previously sent state
/// data. This causes the recipient to ignore all data sent previously.
```

shouldn't this be a doc comment

Reply:
I don't think so. From what I can tell, the generated documentation doesn't include the test files.